ascii.py

```diff
@@ -5,65 +5,62 @@
 from math import floor
 from PIL import Image
 
+TERMINAL_DIMENSIONS = [180, 41, 10, 22]
+NOTEPAD_PLUS_BIG_DIMENSIONS = [400, 100, 10, 18]
+GITHUB_DIMENSIONS = [223, 40, 10, 29]
+DIMENSIONS = GITHUB_DIMENSIONS
+
 def grayscale(img):
-    # Convert img to grayscale.
+    "Convert img to grayscale."
     return img.convert('RGB').convert('L')
 
+
 def average_val(pixels, y, x, y_step, x_step):
-    # Return average value in pixels numpy array in the given rectangle
-    # Expects rectangle coordinates to be in the range of the pixels array
+    """Return average value in boolean pixels numpy array in the given rectangle
+    Expects rectangle coordinates to be in the range of the pixels array."""
     if y_step == 0:
         raise Exception('y_step must be non-zero')
     if x_step == 0:
         raise Exception('x_step must be non-zero')
 
-    start_y = int(floor(y))
-    end_y = int(floor(y+y_step))
-    start_x = int(floor(x))
-    end_x= int(floor(x+x_step))
+    end_y = y + y_step + 1
+    end_x = x + x_step + 1
 
-    total = 0
-    count = 0
+	dark_pixels = pixels[y:end_y, x:end_x].sum()
+    all_pixels = (x_step + 1) * (y_step + 1)
 
-    for py in range(start_y, end_y+1):
-        for px in range(start_x, end_x+1):
-            total += pixels[py, px]
-            count += 1
+    return dark_pixels / all_pixels
 
-    return total / count
 
 def calculate_step_sizes(height, width):
-    # Calculate how big a region of the image each ASCII char should represent,
-    # so that we remain inside the max output dimensions.
-
-    TERMINAL_DIMENSIONS = [180, 41, 10, 22]
-    NOTEPAD_PLUS_BIG_DIMENSIONS = [400, 100, 10, 18]
-    GITHUB_DIMENSIONS = [223, 40, 10, 29]
-    DIMENSIONS = GITHUB_DIMENSIONS
+    """ Calculate how big a region of the image each ASCII char should represent,
+    so that we remain inside the max output dimensions."""
 
     # Max number of output characters to display
-    MAX_OUTPUT_WIDTH = DIMENSIONS[0]
-    MAX_OUTPUT_HEIGHT = DIMENSIONS[1]
+    max_output_width = DIMENSIONS[0]
```

Comment:
non constants should ideally be lowercase, this is just convention, feel free to ignore

Reply:
yup. I do too much Bash where variables tend to all be uppercase, and I still do that as habit for things that seem 'constant-y'